compiler/rustc_hir_typeck/src/cast.rs

```diff
@@ -194,18 +194,47 @@ fn make_invalid_casting_error<'a, 'tcx>(
     )
 }
 
+/// If a cast from `from_ty` to `to_ty` is valid, returns an Ok containing the kind of
+/// the cast. In certain cases, including some invalid casts from array references
+/// to pointers, this may cause additional errors to be emitted and/or ICE error
+/// messages. This function will panic if that occurs.
```

Comment:
Is this really true? I don't see any panicking done in the body of this function, but I assume `CastCheck` wouldn't do that.

Reply:
Oh heh. I didn't read the comment at all and just copied over the function